### ISSUE TITLE
enhance: factor broadcasting loop hangs forever when factor has more dims than the image

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4718]
## Bug

Several `kornia.enhance` functions make `factor` broadcastable with this loop:

```python
while len(factor.shape) != len(image.shape):
    factor = factor[..., None]
```

If `factor` has **more** dimensions than `image`, the loop never ends: each iteration adds a dimension, so the lengths never match. The call hangs forever and never raises.

### Reproduction (current `main`, torch 2.14, CPU)

```python
import torch, kornia
kornia.enhance.adjust_brightness(torch.rand(3, 4, 4), torch.ones(2, 1, 1, 1))  # never returns
```

With a 20 s subprocess timeout, all five of these hang: `adjust_saturation_raw`, `adjust_hue_raw`, `adjust_saturation_with_gray_subtraction`, `adjust_brightness` and `adjust_contrast`. A common trigger is an unbatched `(C, H, W)` image together with a batched factor.

### Occurrences on `main`

```
kornia/enhance/adjust.py:89:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:147:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:219:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:416:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:475:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:550:    while len(factor.shape) != len(image.shape):
kornia/enhance/adjust.py:605:    while len(factor.shape) != len(image.shape):
```

#4717 adds one more copy in the new direct `adjust_saturation` path.

### Suggested fix

Raise a `ValueError` when `factor.dim() > image.dim()`, or when a factor dimension fails to broadcast. Ideally this goes in one shared helper that all the call sites use, with a test per public function.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
